packages/core/workflows-sdk/src/utils/composer/__tests__/compose.ts

```diff
@@ -927,33 +923,35 @@ describe("Workflow composer", function () {
         }
       })
 
-      workflow.changeProduct((theReturnOfThePreviousHook) => {
-        return {
-          ...theReturnOfThePreviousHook,
-          moreProperties: "2nd hook update",
-        }
-      })
+      expect(() =>
+        workflow.hooks.changeProduct((theReturnOfThePreviousHook) => {
+          return {
+            ...theReturnOfThePreviousHook,
+            moreProperties: "2nd hook update",
+          }
+        })
+      ).toThrow(
+        "Cannot define multiple hook handlers for the changeProduct hook"
+      )
 
       const workflowInput = "id_123"
       const { result: final } = await workflow().run({
         input: workflowInput,
       })
 
-      expect(warn).toHaveBeenCalledTimes(1)
       expect(final).toEqual({
         id: "id_123",
-        prod: "product_1**",
-        var: "variant_2**",
+        // prod: "product_1**",
```

Comment:
question: Same question

Reply:
Ohh yeah. Wanted to talk about it with @adrien2p once, because the old hooks were able return new data that gets merged with the output in the test.